hiclass/LocalClassifierPerParentNode.py

```diff
@@ -165,6 +194,126 @@ def predict(self, X):
 
         return y
 
+    def predict_proba(self, X):
+        """
+        Predict class probabilities for the given data.
+
+        Hierarchical labels are returned.
+        If return_all_probabilities=True: Returns the probabilities for each level.
+        Else: Returns the probabilities for the lowest level.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            The input samples. Internally, its dtype will be converted
+            to ``dtype=np.float32``. If a sparse matrix is provided, it will be
+            converted into a sparse ``csr_matrix``.
+        Returns
+        -------
+        T : ndarray of shape (n_samples,n_classes) or List[ndarray(n_samples,n_classes)]
+            The predicted probabilities of the lowest levels or of all levels.
+        """
+        # Check if fit has been called
+        check_is_fitted(self)
+
+        # Input validation
+        if not self.bert:
+            X = check_array(X, accept_sparse="csr", allow_nd=True, ensure_2d=False)
+        else:
+            X = np.array(X)
+
+        if not self.calibration_method:
+            self.logger_.info(
+                "It is not recommended to use predict_proba() without calibration"
+            )
+
+        self.logger_.info("Predicting Probability")
+
+        # Initialize array that holds predictions
+        y = np.empty((X.shape[0], self.max_levels_), dtype=self.dtype_)
+
+        # Predict first level
+        classifier = self.hierarchy_.nodes[self.root_]["classifier"]
+        # use classifier as a fallback if no calibrator is available
+        calibrator = (
+            self.hierarchy_.nodes[self.root_].get("calibrator", classifier)
+            or classifier
+        )
+        proba = calibrator.predict_proba(X)
+
+        y[:, 0] = calibrator.classes_[np.argmax(proba, axis=1)]
```

Comment:
If you need to use the predictions, wouldn't it be better to use the already implemented predict method? I imagine it could simplify the code here and avoid redundancy

Reply:
This is not the same as calling the implemented predict method, because it is using calibrated probabilities for the prediction while the predict method uses uncalibrated probabilities.